web/src/lib/components/faces-page/set-birth-date-modal.svelte

```diff
@@ -18,10 +18,13 @@
 
   $: {
     if (birthDate && birthDate > todayFormatted) {
-      // If the birthDate is set in the future, force the date to today's date
-      birthDate = todayFormatted;
+      // If the birthDate is set in the future, set error message
       error = 'Date of birth cannot be in the future.';
+    } else if (birthDate === '') {
+      // If birthDate is intentionally cleared, reset error and allow empty submission
+      error = '';
     } else {
+      // Reset error for valid dates
       error = '';
     }
   }
```

Comment:
Now that I think about it, do we need to check that the date is a future date if the input prevents it? If not, I don't think we need this `error`

Reply:
The form will prevent the user from selecting a date in the future, as it will have max(date)=today. But This check is more if the user type it. 
I don't know how the component will behave if the user type 22.02.2030, will it just disable the set button ? or will it force the date to be today ? As I am not sure, I did all those checks. Better safe than sorry

Reply:
Don't worry for the error. In the worst scenario, the user will have a notification which explain what's the error.

This is the user flow with : 
```ts
export let birthDate: string | null;
```
```svelte
  const handleSubmit = () => {
    if (!birthDate) {
      return;
    }
    dispatch('updated', birthDate);
  };
```
```svelte
<Button type="submit" fullwidth disabled={!birthDate}>Set</Button>
```
https://github.com/immich-app/immich/assets/74269598/b130bb97-185f-47b5-a5f5-8cebcf5a08fd

Reply:
Ok I will do this changes, 
It make the code cleaner. 
But be aware of this corner case : 
If the user set a date of brith correct and save it, and for some reasons (s)he decide to delete the birth date of that person, I guess it will not be possible anymore as the birth date will be empty

Reply:
Oh yes, I completely forgot that.
In this case, you just want to be sure that the date entered by the user is different from the original one.